executor/compiler.go

```diff
@@ -46,17 +49,166 @@ func (c *Compiler) Compile(goCtx goctx.Context, stmtNode ast.StmtNode) (*ExecStm
 		return nil, errors.Trace(err)
 	}
 
+	CountStmtNode(stmtNode, c.Ctx.GetSessionVars().InRestrictedSQL)
+	isExpensive := logExpensiveQuery(stmtNode, finalPlan)
+
 	return &ExecStmt{
 		InfoSchema: infoSchema,
 		Plan:       finalPlan,
-		Expensive:  stmtCount(stmtNode, finalPlan, c.Ctx.GetSessionVars().InRestrictedSQL),
+		Expensive:  isExpensive,
 		Cacheable:  plan.Cacheable(stmtNode),
 		Text:       stmtNode.Text(),
 		StmtNode:   stmtNode,
 		Ctx:        c.Ctx,
 	}, nil
 }
 
+func logExpensiveQuery(stmtNode ast.StmtNode, finalPlan plan.Plan) (expensive bool) {
+	expensive = isExpensiveQuery(finalPlan)
+	if !expensive {
+		return
+	}
+
+	const logSQLLen = 1024
+	sql := stmtNode.Text()
+	if len(sql) > logSQLLen {
+		sql = fmt.Sprintf("%s len(%d)", sql[:logSQLLen], len(sql))
+	}
+	log.Warnf("[EXPENSIVE_QUERY] %s", sql)
```

Comment:
I'm afraid there would too many EXPENSIVE_QUERY after this change.

Reply:
any suggestion ?